test/ILLink.RoslynAnalyzer.Tests/TestCaseUtils.cs

```diff
@@ -95,6 +121,38 @@ public static void RunTest<TAnalyzer> (MemberDeclarationSyntax m, List<Attribute
 			return builder.ToImmutable ();
 		}
 
+		public static ImmutableDictionary<string, List<string>> GetReferenceFilesByDirName ()
+		{
+			var builder = ImmutableDictionary.CreateBuilder<string, List<string>> ();
+
+			foreach (var file in GetReferenceFiles ()) {
+				var dirName = Path.GetFileName (Path.GetDirectoryName (file))!;
+				if (builder.TryGetValue (dirName, out var sources)) {
+					sources.Add (file);
+				} else {
+					sources = new List<string> () { file };
+					builder[dirName] = sources;
+				}
+			}
+
+			return builder.ToImmutable ();
+		}
+
+		public static IEnumerable<string> GetReferenceFiles ()
```

Comment:
This is only called by `GetReferenceFilesByDirName`, should we collapse these two methods into just one?